### PR TITLE
Add PBP Test Suite

### PR DESCRIPTION
**Pull request by mringwal**
This PR adds the PBP layer and WIDs for the PBP Test suite

It also fixes smaller details in other areas we've stumbled upon using one commit per fix.

PTS Project PBP code: https://github.com/bluekitchen/auto-pts/blob/develop-le-audio-tmap/autopts/ptsprojects/btstack/pbp.py

### CODE REVIEW

**Comment by mringwal** [comment]
Ping. Is there anything from my side to move forward with this?

**Comment by sjanc** [comment]
#AutoPTS run zephyr TBS GTBS BAP/BA/CGGIT TMAP

**Comment by mringwal** [comment]
Looks like I forgot to git add layers/pbp.py. Added now.

**Comment by sjanc** [comment]
doc/btp_pbp.txt seems also be missing

BTW first 3 commits with bugfixes could go in immediately if you send them in separate PR

**Comment by mringwal** [comment]
Ok. Didn't know what's better/more efficient. I'll go for more concise PRs then.

**Comment by mringwal** [comment]
Well. Looks like all bugfixes got invalid as the corresponding files have vanished/are not there anymore :(
Will get back to this later today.

**Comment by sjanc** [comment]
This PR doesn't show any merge conflicts so...  are you sure you check against right branch/commit ?:)

**Comment by mringwal** [comment]
Yupp, looks like I was looking at something else - but I'm glad it was straight forward. Ok, new PR with the bug fixes: https://github.com/auto-pts/auto-pts/pull/1245

**Comment by mringwal** [comment]
Since the bug fixes have been merged, I've rebased this PR and it looks ready for integration to me :)

File: doc/btp_pbp.txt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		than required bytes received) it shall be assumed that command
+		is not supported.
+
+		In case of an error, the error response will be returned.
+
+	Opcode 0x02: Set Public Broadcast Announcement
+
+		Controller Index:	<controller id>
+		Command parameters:	Features (1 octet)
+							Metadata LTVs len (1 octet)
```
Comment (sjanc) [inline doc/btp_pbp.txt:27]: I wonder if those should be passed by upper tester,  what exactly is pass here?  eg name is configured with separate command

Comment (mringwal) [reply doc/btp_pbp.txt:27]: in Set Public Broadcast Announcement, the service data for ORG_BLUETOOTH_SERVICE_PUBLIC_BROADCAST_ANNOUNCEMENT is set. It contains metadata.

In Set Broadcast Name, ad type BLUETOOTH_DATA_TYPE_BROADCAST_NAME is set with the name, ie. the name is not in the Set Public Broadcast Announcement meta data.

That being set, it looks like we never call Set Broadcast Name and it's not set by default in our tester implementation. As it is an official field, I would suggest to keep it and improve the documentation a bit if that's ok with you.

**Pull request merged**